apps/webapp/shared/input-token/select-token-modal.tsx

```diff
@@ -0,0 +1,72 @@
+'use client';
+
+import React, { Dispatch, SetStateAction, useMemo, useState } from 'react';
+import { Dialog, DialogContent, DialogHeader, DialogPrimitive, DialogTrigger, Input } from 'ui';
+import { ResponsiveImage } from '../responsive-image';
+import { Asset } from '../../app/send/types';
+import { MagnifyingGlassIcon } from '@radix-ui/react-icons';
+import { assets } from '../../app/send/constants';
+
+interface SelectTokenModalProps {
+  asset: Asset;
+  setAsset: Dispatch<SetStateAction<Asset>>;
+}
+
+export const SelectTokenModal = ({ asset, setAsset }: SelectTokenModalProps) => {
+  const [search, setSearch] = useState('');
+
+  const filteredAsset = useMemo(() => {
+    if (!search) return assets;
+    return assets.filter(asset => asset.name.toLowerCase().includes(search.toLowerCase()));
+  }, [search]);
+
+  return (
+    <Dialog>
+      <DialogTrigger asChild>
+        <div className='flex items-center gap-2 bg-light-brown px-6 py-[6px] rounded-lg cursor-pointer'>
+          <ResponsiveImage src={asset.icon} alt='Asset' className='w-6 h-6' />
+          <p className='text-base font-bold text-light-grey'>{asset.name}</p>
+        </div>
+      </DialogTrigger>
+      <DialogContent className='max-w-[400px] bg-charcoal-secondary'>
+        <div className='relative z-10 gap-4 flex flex-col pb-5'>
+          <DialogHeader className='border-b'>Select asset</DialogHeader>
+          <div className='flex flex-col gap-4 px-[30px]'>
+            <div className='relative flex w-full items-center justify-center gap-4'>
+              <div className='absolute inset-y-0 left-3 flex items-center'>
+                <MagnifyingGlassIcon className='h-5 w-5 text-muted-foreground' />
+              </div>
+              <Input
+                className='pl-10'
+                value={search}
+                onChange={e => setSearch(e.target.value)}
+                placeholder='Search asset...'
+              />
+            </div>
+            <div className='flex justify-between items-center font-headline text-base font-semibold mt-2'>
+              <p>Token name</p>
+              <p>Balance</p>
+            </div>
+            <div className='flex flex-col gap-2'>
+              {filteredAsset.map(asset => (
+                <DialogPrimitive.Close key={asset.name}>
```

Comment:
This appears to have left/right margin. Think it can extend to the end.

Reply:
ok, but it from figma